docs/proposals/apiman-integration.md

```diff
@@ -0,0 +1,132 @@
+      api.service.kubernetes.io/description-language: SwaggerJSON
+```
+Additionally, services that should be managed by APIMan will be further annotated identifying who is managing the service.  The proposed annotation is:
+```
+  openshift.io/service-managed-by: apiman
+```
+
+Service providers will need to explicitly publish a service using the APIMan user interface.  Future iterations may include functionality to automatically publish a service when these annotations are applied.
+
+### Origin Web Console UI extension
+A custom extension<sup>[6](#r6)</sup> will be created and hooked into the web console to support API endpoints.  The extension
+will allow a user to navigate to the APIMan management interface to:
+* Manage an API endpoint
+* Expose an annotated service as a new API endpoint
+
+The extension will provide the following details to the APIMan gateway:
+* Service name
+* Service namespace
+* User's Oauth token
+* Back link to the openshift web console
+
+Calls from the extension to the APIMan management interface will utilize a REST POST call where the oauth token is part of the payload.  It will
+utilize a similiar design<sup>[7](#r7)</sup> that is realized by the OpenShift origin aggregated logging integration and the auth proxy<sup>[8](#r8)</sup>.  It is necessary for
+security reasons to not add the token as a query parameter.  
+
+### API Services Gateway
+The APIMan gateway will be deployed as a cluster-wide infrastructure component to handle services exposed as API endpoints.  Internal and external clients can use the gateway to access APIs.  Admins can configure the cluster to limit direct access to services and encourage consumers of API endpoints to utilize the gateway by deploying the multi-tenant SDN plugin.  This plugin controls service to service communication between projects and would force traffic throught the gateway.  A route will be created as a well known endpoint to the APIMan gateway through which all API services traffic will flow.
+
+#### Service Consumption & Authentication
+The details of service authentication and consumption depend upon the type of published service and the polices associated with it. Essentially, user's of public service need to simply fullfill the policies associated with the service.  User's of contracted services will provide an APIMan generated API token with their request.  These are fundamental features of APIMan and are described here to provide a simple understanding of consuming APIMan controlled services.  Additional information can be found in the [APIMan](http://www.apiman.io/) documentation.
+
+### OpenShift CLI Modifications
+Modifications to the openshift client binary are a stretch goal for the initial integration.  It will be updated to allow a user to:
+* Expose an API endpoint.  
+Exposing the service will update the service annotation as described by the service annotation section.  Possible usage syntax:<p>
+``` oc set api-service SERVICENAME --path PATH```<p>
```

Comment:
These modifications are for the initial release syntactic sugar for adding the annotations.  Potentially, in future, the action of setting these annotations would expose your service based upon the default policies laid out by your organization.  Presumably it could provide the ui link where you could actively modify them as well.